### PR TITLE
Tasks 2.1 - 2.5

### PR DESCRIPTION
**Pull request by mikeivashka**

### CODE REVIEW

File: core/src/main/java/com/es/core/cart/HttpSessionCartService.java [hunk 1]
```diff
@@ -1,28 +1,72 @@
...
+
+    @Resource
+    private PhoneDao phoneDao;
+
     @Override
-    public Cart getCart() {
-        throw new UnsupportedOperationException("TODO");
+    public Map<Phone, Long> getCart() {
+        Map<Phone, Long> cartItemsMap = new HashMap<>();
+        cart.getCartItems().forEach(item -> cartItemsMap.put(item.getPhone(), item.getQuantity()));
```
Comment (marty-friedman) [inline core/src/main/java/com/es/core/cart/HttpSessionCartService.java]: Can be replaced with .collect(Collectors.toMap())

File: core/src/main/java/com/es/core/model/phone/JdbcPhoneDao.java [hunk 2]
```diff
@@ -1,27 +1,73 @@
...
-import java.util.Optional;
+import java.util.*;
 
 @Component
-public class JdbcPhoneDao implements PhoneDao{
+public class JdbcPhoneDao implements PhoneDao {
+    private static final String SQL_GET_COLORS_QUERY = "SELECT colors.id, colors.code FROM colors JOIN phone2color ON colors.id = phone2color.colorId WHERE phoneId = %d";
+    private static final String SQL_GET_PHONE_BY_ID_QUERY = "SELECT * FROM phones WHERE id = %d";
+    private static final String SQL_ALL_PRODUCTS = "SELECT * FROM phones ORDER BY %s %s OFFSET %d LIMIT %d";
+    private static final String SQL_FIND_PRODUCTS_NONNULL_PRICE_POSITIVE_STOCK = "SELECT * FROM phones JOIN stocks ON phones.id = stocks.phoneId WHERE price IS NOT NULL AND stock > 0 AND LOWER(model) LIKE LOWER(?) ORDER BY %s %s OFFSET %d LIMIT %d";
```
Comment (marty-friedman) [inline core/src/main/java/com/es/core/model/phone/JdbcPhoneDao.java]: It works when you specify LOWER('%%%s%%') - don't forget the quotes, as jdbcTemplate no longer inserts them for you when using String#format way. However, I think I'd personally stick to JDBC-args way everywhere in the class, it's a clearer use of the provided API than manually constructing SQL strings in my opinion. Perhaps, you can even try to look at [NamedParameterJdbcTemplate](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/jdbc/core/namedparam/NamedParameterJdbcTemplate.html) to make the code even more pretty and readable.

File: core/src/main/resources/context/applicationContext-core.xml [hunk 3]
```diff
@@ -7,20 +7,20 @@
...
     </bean>
 
     <bean id="jdbcTemplate" class="org.springframework.jdbc.core.JdbcTemplate">
-        <property name="dataSource" ref="dataSource" />
+        <property name="dataSource" ref="dataSource"/>
     </bean>
 
-    <jdbc:initialize-database enabled="true" data-source="dataSource">
-        <jdbc:script location="classpath:db/schema.sql" />
+    <jdbc:initialize-database enabled="${db.insertSchema}">
```
Comment (marty-friedman) [inline core/src/main/resources/context/applicationContext-core.xml:23]: Currently, it cannot be configured via properties whether to import demo data or not for normal context. Please check the spring configuration under demodata module.

File: core/src/test/resources/conf/application-test.properties [hunk 4]
```diff
@@ -0,0 +1,6 @@
+db.driver=org.h2.Driver
+db.url=jdbc:h2:~/phoneshop/db/h2
```
Comment (marty-friedman) [inline core/src/test/resources/conf/application-test.properties]: Regarding in-memory configuration, please see https://stackoverflow.com/a/51824325

File: core/src/main/java/com/es/core/cart/HttpSessionCartService.java (hunk 1 above)
Comment (mikeivashka) [reply core/src/main/java/com/es/core/cart/HttpSessionCartService.java]: Fixed

File: core/src/test/resources/conf/application-test.properties (hunk 4 above)
Comment (mikeivashka) [reply core/src/test/resources/conf/application-test.properties]: Fixed, thanks a lot!

File: core/src/main/resources/context/applicationContext-core.xml (hunk 3 above)
Comment (mikeivashka) [reply core/src/main/resources/context/applicationContext-core.xml:23]: Ooops... 
Fixed

**Pull request merged**